contracts/base/DAIPool.sol

```diff
@@ -0,0 +1,139 @@
+
+
+contract DAIPool is DAIPoolInterface, Initializable {
+    using AddressLib for address;
+
+    /* State Variables */
+
+    IERC20 public dai;
+
+    ZDAIInterface public zdai;
+
+    LenderInfoInterface public lenderInfo;
+
+    LoansInterface public loanInfo;
+
+    /** Modifiers */
+
+    /* Constructor */
+
+    /** External Functions */
+
+    function initialize(
+        address zdaiAddress,
+        address daiAddress,
+        address lenderInfoAddress,
+        address loanInfoAddress
+    ) external isNotInitialized() {
+        zdaiAddress.requireNotEmpty("ZDai address is required.");
+        daiAddress.requireNotEmpty("DAI address is required.");
+        lenderInfoAddress.requireNotEmpty("LenderInfo address is required.");
+        loanInfoAddress.requireNotEmpty("LoanInfo address is required.");
+
+        initialize();
+
+        zdai = ZDAIInterface(zdaiAddress);
+        dai = IERC20(daiAddress);
+        lenderInfo = LenderInfoInterface(lenderInfoAddress);
+        loanInfo = LoansInterface(loanInfoAddress);
+    }
+
+    /**
+        @notice It allows users to deposit DAIs into the pool.
+        @dev the user must call DAI.approve function previously.
+        @param amount of DAIs to deposit in the pool.
+    */
+    function depositDai(uint256 amount) external isInitialized {
+        // Require DAI approval from sender to this contract.
+        requireDAIAllowance(msg.sender, amount);
+
+        // Transfering DAI tokens to DAIPool
+        daiTransferFrom(msg.sender, amount);
+
+        // Mint ZDAI tokens
+        zdaiMint(msg.sender, amount);
+
+        // Notify ZDAI tokens were minted
+        lenderInfo.zDaiMinted(msg.sender, amount);
+
+        // Emit event
+        emit DaiDeposited(msg.sender, amount);
+    }
+
+    function withdrawDai(uint256 amount) external {}
+
+    function repayDai(uint256 amount, address borrower) external {}
+
+    function liquidationPayment(uint256 amount, address liquidator) external {}
+
+    function createLoan(uint256 amount, address borrower) external returns (bool) {}
+
+    function withdrawInterest(uint256 amount) external {}
+
+    /** Internal functions */
+
+    /** Private functions */
+
+    /**
+
+     */
+    function requireDAIAllowance(address owner, uint256 amount) private view {
+        uint256 allowanceResult = dai.allowance(owner, address(this));
+        require(allowanceResult >= amount, "Not enough tokens allowed.");
+    }
+
+    /**
+        @notice It transfers an amount of DAI tokens from an address to this contract.
+        @param from address where the DAI tokens will transfer from.
+        @param amount to be transferred.
+        @dev It throws a require error if 'transferFrom' invocation fails.
+     */
+    function daiTransferFrom(address from, uint256 amount) private {
+        bool transferFromResult = dai.transferFrom(from, address(this), amount);
+        require(transferFromResult, "Transfer from was not successful.");
```

Comment:
dai.transferFrom already fails if the transfer is not successful - you do not need to check this :)

Reply:
We agreed that checking the bool on transferFrom and mint is good in case we add other tokens in future